umibukela/results.py

```diff
@@ -0,0 +1,191 @@
+    """
+    path = path or []
+    question_results = question_results or {}
+    idx = 0  # The index of the child in the form in its array of siblings
+    for child in children:
+        deeper_path = path + [child['name']]
+        if deeper_path in [['facility'], ['demographics_group', 'gender']]:
+            pass
+        elif child_is_type(child, 'group') and child['name'] == 'meta':
+            pass
+        elif child_is_type(child, 'group'):
+            question_results = count_options(
+                submissions,
+                child['children'],
+                deeper_path,
+                question_results,
+                None)
+        elif (child_is_type(child, 'select one')
+              or child_is_type(child, 'select all that apply')):
+            # multiple choice questions
+            current_option_counts = count_question_options(
+                submissions,
+                deeper_path
+            )
+            question_results = deep_dict_set(
+                question_results,
+                child['label'],
+                [pathstr(deeper_path), 'label']
+            )
+            question_results = count_options(
+                submissions,
+                child['children'],
+                deeper_path,
+                question_results,
+                current_option_counts
+            )
+        elif ('type' not in child):
+            # option in multiple choice question
+            question_results = deep_dict_set(
+                question_results,
+                child['label'],
+                [pathstr(path), 'options', child['name'], 'label']
+            )
+            question_results = deep_dict_set(
+                question_results,
+                idx,
+                [pathstr(path), 'options', child['name'], 'idx']
+            )
+            question_results = set_option_counts(
+                path,
+                child['name'],
+                question_results,
+                current_option_counts
+            )
+        else:
+            pass
+        idx += 1
+    return question_results
+
+
+def child_is_type(child, type):
+    return ('type' in child) and child['type'] == type
+
+
+def count_question_options(site_submissions, path):
+    cols = ['facility', 'demographics_group/gender', pathstr(path)]
+    question_table = site_submissions.loc[:, cols]
+    question_counts = question_table.groupby(
+        ['demographics_group/gender', pathstr(path)]
+    ).count()
+    return question_counts
+
+
+def set_option_counts(path, option_name, results, current_option_counts):
+    for gender in ['male', 'female']:
+        option_table = current_option_counts
+
+        try:
+            val = int(option_table.loc[gender, option_name])
+        except KeyError:
+            # values that aren't counted because they don't occur in the
+            # results for this question won't be indexes in the counts
+            print("########################################")
```

Comment:
It's better to use a logger than print, since we can then control whether or not it's shown.

```python
from logging import getLogger

# a module-level logger is fine
log = getLogger(__name__)
...
log.info("...")
```